specifyweb/frontend/js_src/lib/components/WorkBench/WbActions.tsx

```diff
@@ -15,7 +15,186 @@ import type { WbCellCounts } from './CellMeta';
 import { RollbackConfirmation } from './Components';
 import { CreateRecordSetButton } from './RecordSet';
 import { WbStatus as WbStatusComponent } from './Status';
-import type { WbStatus, WbView } from './WbView';
+import type { DialogHandlers, WbStatus, WbView } from './WbView';
+import type { Dataset } from '../WbPlanView/Wrapped';
+import type { WbMapping } from './mapping';
+import type { RA } from '../../utils/types';
+
+/* eslint-disable functional/no-this-expression */
+export class WbActionsReact {
+  // eslint-disable-next-line functional/prefer-readonly-type
+  public hasUnSavedChanges: boolean = false;
+
+  public constructor(
+    private readonly data: RA<RA<string | null>>,
+    private readonly dataset: Dataset,
+    private readonly mappings: WbMapping,
+    private readonly noUploadPlan: DialogHandlers,
+    private readonly uploadDialog: DialogHandlers,
+    private readonly statusComponent: DialogHandlers,
+    private readonly checkDeletedFail: (statusCode: number) => boolean,
+    private readonly triggerRefresh: () => void,
+    private readonly saveProgressBar: DialogHandlers,
```

Comment:
This constructor is starting to look crazy and is only going to get worse. Would appreciate some suggestions for improvement.